python/cudf/cudf/core/column/column.py

```diff
@@ -395,16 +395,23 @@ def view(self, dtype):
             The dtype to view the data as
         
         """
-        if self.null_count > 0:
-            raise ValueError("Can not produce a view of a column with nulls")
         dtype = np.dtype(dtype)
+
+        if self.null_count > 0:
+            if self.dtype.itemsize != dtype.itemsize:
+                raise ValueError("Can not produce a view of a column with nulls")
```

Comment:
Can we make this error more explicit that we can't produce a view of a column with nulls of a different size dtype?

Reply:
And should we also handle things where element is `NAN` and still valid ?

Reply:
Is there anything wrong with just letting it get converted to whatever the binary representation happens to translate to? That appears to be what happens with NumPy.

Reply:
Nope, that's just the nature of binary representations :)